### PR TITLE
Support Node shebang scripts

### LINKED ISSUE #69: File with shebang doesn't work
The first script I tried ncc on failed like this:

```
(node:71332) UnhandledPromiseRejectionWarning: Error: Hash: c93a3aedc91c7aab0770
Version: webpack 4.26.0
Time: 127ms
Built at: 2018-11-27 16:04:22
 Asset     Size  Chunks  Chunk Names
out.js  4.7 KiB       0  main
Entrypoint main = out.js
[0] ./2-stats2boundaries.js 1.11 KiB {0} [built] [failed] [1 error]

ERROR in ./2-stats2boundaries.js
Module build failed (from (webpack)/ncc/asset-relocator.js):
SyntaxError: Unexpected character '#' (1:0)
    at Parser.module.exports.pp$4.raise (/Users/stevebennett/npm/lib/node_modules/@zeit/ncc/dist/ncc/asset-relocator.js:16125:13)
...
```

The first line of that script is:

`#!/usr/bin/env node  --max-old-space-size=4096`

### PR DESCRIPTION
**Pull request by guybedford**
Fixes #69.

### CODE REVIEW

File: test/unit/shebang/input.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/user/env
```
Comment (TooTallNate) [inline test/unit/shebang/input.js]: `#!/bin/user/env` -> `#!/bin/user/env node` ?

File: test/unit/shebang/output.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
+module.exports =
```
Comment (TooTallNate) [inline test/unit/shebang/output.js]: Should the shebang be copied to the output entrypoint file?

Comment (TooTallNate) [reply test/unit/shebang/output.js]: And also possibly setting executable permissons (or copy the permissons from the entrypoint)?

**Comment by guybedford** [comment]
I've added the features of #107 to this PR, without making the shebang explicit in the API and instead just passing it through in the code source as suggested.

**Comment by rauchg** [comment]
@guybedford can you re-base please?

**Comment by guybedford** [comment]
Rebased.

**Pull request merged**